examples/MMF-activity-recognition/handler.py

```diff
@@ -0,0 +1,140 @@
+        df = pd.read_csv('./charades_action_lables.csv')
+        label_set = set()
+        df['action_labels'] = df['action_labels'].str.replace('"','')
+        labels_initial = df['action_labels'].tolist()
+        labels = []
+        for sublist in labels_initial:
+            new_sublist = ast.literal_eval(sublist)
+            labels.append(new_sublist)
+            for item in new_sublist:
+                label_set.add(item)
+        classes = sorted(list(label_set))
+        self.class_to_idx = {classes[i]: i for i in range(len(classes))}
+        self.classes = classes
+        self.labels = labels
+        self.idx_to_class = classes
+        config = OmegaConf.load('config.yaml')
+        print("*********** config keyssss **********", config.keys())
+        setup_very_basic_config()
+
+        setup_imports()
+        self.model = MMFTransformer(config.model_config.mmf_transformer)
+        self.model.build()
+        self.model.init_losses()
+        self.processor = build_processors(
+            config.dataset_config["charades"].processors
+        )
+        state_dict = torch.load(serialized_file)
+        self.model.load_state_dict(state_dict)
+        self.model.to(self.device)
+        self.model.eval()
+        self.initialized = True
+        print("********* files in temp direcotry that .mar file got extracted *********", os.listdir(model_dir))
+
+    def preprocess(self, requests):
+        """ Preprocessing, based on processor defined for MMF model.
+            """
+
+        def create_sample(video_transfomred,audio_transfomred,text_tensor, video_label):
+
+            label = [self.class_to_idx[l] for l in video_label]
+
+            one_hot_label = torch.zeros(len(self.class_to_idx))
+            one_hot_label[label] = 1
+
+            current_sample= Sample()
+            current_sample.video = video_transfomred
+            current_sample.audio = audio_transfomred
+            current_sample.update(text_tensor)
+            current_sample.targets = one_hot_label.to(self.device)
+            current_sample.dataset_type = 'test'
+            current_sample.dataset_name = 'charades'
+            return SampleList([current_sample])
+
+        for idx, data in enumerate(requests):
+            raw_script = data.get('script')
+            script = raw_script.decode('utf-8')
+            raw_label = data.get('lables')
+            video_label = raw_label.decode('utf-8')
+            video_label = [video_label]
+            
+            video = io.BytesIO(data['data'])
+            video_tensor, audio_tensor,info = torchvision.io.read_video(video)
+            text_tensor = self.processor["text_processor"]({"text": script})
+            video_transformed = self.processor["video_test_processor"](video_tensor)
+            audio_transformed = self.processor["audio_processor"](audio_tensor)
+            samples = create_sample(video_transformed,audio_transformed,text_tensor,video_label)
```

Comment:
Not a review comment but it's cool how simple the library makes this kind of work

Reply:
@msaroufim MMF team did a great job on that.